generator/integration_tests/tests/request_id_connection_impl_test.cc

```diff
@@ -0,0 +1,281 @@
+#include "google/cloud/grpc_options.h"
+#include "google/cloud/options.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include <gmock/gmock.h>
+#include <memory>
+#include <set>
+#include <string>
+#include <utility>
+
+namespace google {
+namespace cloud {
+namespace golden_v1 {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using ::google::cloud::testing_util::IsOkAndHolds;
+using ::google::test::requestid::v1::CreateFooRequest;
+using ::google::test::requestid::v1::Foo;
+using ::google::test::requestid::v1::ListFoosRequest;
+using ::google::test::requestid::v1::ListFoosResponse;
+using ::google::test::requestid::v1::RenameFooRequest;
+using ::testing::_;
+using ::testing::ByMove;
+using ::testing::ElementsAre;
+using ::testing::Eq;
+using ::testing::IsEmpty;
+using ::testing::ResultOf;
+using ::testing::Return;
+
+class MockRequestIdServiceStub
+    : public google::cloud::golden_v1_internal::RequestIdServiceStub {
+ public:
+  ~MockRequestIdServiceStub() override = default;
+
+  MOCK_METHOD(StatusOr<google::test::requestid::v1::Foo>, CreateFoo,
+              (grpc::ClientContext&,
+               google::test::requestid::v1::CreateFooRequest const&),
+              (override));
+
+  MOCK_METHOD(future<StatusOr<google::longrunning::Operation>>, AsyncRenameFoo,
+              (google::cloud::CompletionQueue&,
+               std::shared_ptr<grpc::ClientContext>, Options const&,
+               google::test::requestid::v1::RenameFooRequest const&),
+              (override));
+
+  MOCK_METHOD(StatusOr<google::test::requestid::v1::ListFoosResponse>, ListFoos,
+              (grpc::ClientContext&,
+               google::test::requestid::v1::ListFoosRequest const&),
+              (override));
+
+  MOCK_METHOD(future<StatusOr<google::test::requestid::v1::Foo>>,
+              AsyncCreateFoo,
+              (google::cloud::CompletionQueue&,
+               std::shared_ptr<grpc::ClientContext>,
+               google::test::requestid::v1::CreateFooRequest const&),
+              (override));
+
+  MOCK_METHOD(future<StatusOr<google::longrunning::Operation>>,
+              AsyncGetOperation,
+              (google::cloud::CompletionQueue&,
+               std::shared_ptr<grpc::ClientContext>, Options const&,
+               google::longrunning::GetOperationRequest const&),
+              (override));
+
+  MOCK_METHOD(future<Status>, AsyncCancelOperation,
+              (google::cloud::CompletionQueue&,
+               std::shared_ptr<grpc::ClientContext>, Options const&,
+               google::longrunning::CancelOperationRequest const&),
+              (override));
+};
+
+template <typename Request>
+auto WithRequestId(std::string expected) {
+  return ResultOf(
+      "request has request_id",
+      [](Request const& request) { return request.request_id(); },
+      Eq(std::move(expected)));
+}
+
+template <typename Request>
+auto WithRequestId() {
```

Comment:
This seems backwards. `WithoutRequestId()` ?

Reply:
Looking ahead, seems like you will want a `WithGeneratedRequestId()` matcher that verifies that the ConnectionImpl supplies a UUID for the request_id.

Reply:
> This seems backwards. `WithoutRequestId()` ?

Done.

> Looking ahead, seems like you will want a `WithGeneratedRequestId()` matcher that verifies that the ConnectionImpl supplies a UUID for the request_id.

I was going to go with "not empty", but sure, will do that.